### ISSUE TITLE
apsim-server READ bug

### ISSUE THREAD

**Issue by d11638104** [APSIMInitiative/ApsimX#8771]
### What happened?

Starting server...
Listening on 0.0.0.0:27746...
Waiting for connections...
Client connected to server.
Waiting for commands...
Received RunCommand with 1 changes
Raw job took 651ms
Command ran in 1679ms
Waiting for commands...
Received ReadCommand with 1 parameters
Command ran in 1821ms
Waiting for commands...

When debugging my apsim-server in Visual Studio, my arguments are -f C:/Users/fandy/Desktop/Yang/Yang/test.apsimx -v -k -n -r -a 0.0.0.0.
![屏幕截图 2024-03-20 154308](https://github.com/APSIMInitiative/ApsimX/assets/31063625/21e00f35-47ab-4f0f-97c0-744470a6d897)

 I used the Apsim.Client as mentioned in the tutorial and successfully debugged it. After running, I could see the results in the debug interface. However, after reading, I am unable to obtain the results I saw. I only received a System.double, which is quite perplexing.
![屏幕截图 2024-03-20 154408](https://github.com/APSIMInitiative/ApsimX/assets/31063625/84078129-01d1-4e89-a311-cecaf0cca327)

### What Operating System are you on?

Windows

**Comment by peter-devoil** [comment]
My apologies - I never committed changes made to the data exchange protocol. You'll see them eventually merged [here](https://github.com/APSIMInitiative/APSIM.Client/pull/1). Let us know how you go.

**Comment by d11638104** [comment]
> My apologies - I never committed changes made to the data exchange protocol. You'll see them eventually merged [here](https://github.com/APSIMInitiative/APSIM.Client/pull/1). Let us know how you go.

You are correct. I have tried numerous methods and carefully examined the source code of ApsimServer, and it has not been altered.
